lib/src/facts/linux/filesystem_resolver.cc

```diff
@@ -83,97 +112,139 @@ namespace facter { namespace facts { namespace linux {
 
     void filesystem_resolver::collect_partition_data(data& result)
     {
-        // Only return partition data if we're using libblkid
-#ifdef USE_BLKID
-        // The size of a block, in bytes, read in from /sys/class/block
-        const int block_size = 512;
-
-        // Get a default cache
-        blkid_cache cache;
-        if (blkid_get_cache(&cache, nullptr) < 0) {
-            LOG_ERROR("blkid_get_cache failed: %1% (%2%): partition data is unavailable.", strerror(errno), errno);
-            return;
-        }
-
-        // Probe all block devices
-        if (blkid_probe_all(cache) < 0) {
-            blkid_put_cache(cache);
-            LOG_ERROR("blkid_probe_all failed: %1% (%2%): partition data is unavailable.", strerror(errno), errno);
-            return;
+        // Populate a map of device -> mountpoint
+        map<string, string> mountpoints;
+        for (auto const& point : result.mountpoints) {
+            mountpoints.insert(make_pair(point.device, point.name));
         }
 
-        // Begin iteration
-        auto iter = blkid_dev_iterate_begin(cache);
-        if (!iter) {
-            blkid_put_cache(cache);
-            LOG_ERROR("blkid_dev_iterate_begin failed: %1% (%2%): partition data is unavailable.", strerror(errno), errno);
-            return;
-        }
+        void* cache = nullptr;
 
-        // Populate a map of device -> mountpoint
-        map<string, string> device_mountpoints;
-        for (auto const& point : result.mountpoints) {
-            device_mountpoints.insert(make_pair(point.device, point.name));
+#ifdef USE_BLKID
+        blkid_cache actual = nullptr;
+        if (blkid_get_cache(&actual, nullptr) == 0) {
+            cache = actual;
+        } else {
+            LOG_DEBUG("blkid_get_cache failed: partition attributes are not unavailable.");
```

Comment:
Looks like I edited the string at a later time and didn't remove the `un`.  Will fix.